fem/bilininteg.hpp

```diff
@@ -2094,15 +2106,31 @@ class DiffusionIntegrator: public BilinearFormIntegrator
    Vector pa_data;
    bool symmetric = true; ///< False if using a nonsymmetric matrix coefficient
 
+   // Data for NURBS patch PA
+   int numPatches = 0;
+   std::vector<std::vector<std::vector<std::vector<Vector>>>> reducedWeights;
+   std::vector<std::vector<std::vector<std::vector<std::vector<int>>>>> reducedIDs;
+   std::vector<Array<int>> pQ1D, pD1D;
+   std::vector<std::vector<Array2D<double>>> pB, pG;
+   std::vector<std::vector<std::vector<int>>> pminD, pmaxD, pminQ, pmaxQ, pminDD,
+       pmaxDD;
```

Comment:
It might be good to define typedefs (within the class) for these nested containers. It would be more readable. Also, it would provide an opportunity to add explanations of the indices being used e.g. "this nested container takes four indices which are ...".

Reply:
I defined a typedef to simplify.